src/content_script.js

```diff
@@ -179,120 +173,228 @@ function createBadgeFragment (socialAction) {
+  const xLeft = fencePosition.x - iframePaddingAllowance;
+  const yPos = fencePosition.y - offsetY;
+
+  // Mobile Values
+  const xPosMobile = fencePosition.x;
+  const yPosMobile = fencePosition.y + offsetY;
 
-  // Create Prompt/Allow Dialog
-  if (socialAction === "login"){
-    const htmlBadgeFragmentPromptDiv = htmlBadgeFragment.querySelector(".fbc-badge-prompt");
+  const iconRightAllowance = window.innerWidth - fencePosition.x + fencePos.offsetWidth;
+  const iconLeftAllowance = window.innerWidth - iconRightAllowance;
 
-    const htmlBadgeFragmentPromptH1 = document.createElement("h1");
+  // Desktop Orientation
+  if (iconRightAllowance > iframePaddingAllowance || iconLeftAllowance > iframePaddingAllowance) {
 
-    htmlBadgeFragmentPromptH1.appendChild(document.createTextNode( browser.i18n.getMessage("facebookContainer") ));
-    htmlBadgeFragmentPromptDiv.appendChild(htmlBadgeFragmentPromptH1);
+    // Position iframe relative to FBC Icon
+    iframeBox.style.marginLeft = `${xRight}px`;
+    iframeBox.style.marginTop = `${yPos}px`;
 
-    const htmlBadgeFragmentPromptContents = document.createElement("div");
-    htmlBadgeFragmentPromptContents.className = "fbc-badge-prompt-contents";
+    // Add Chevron (Default left arrow)
+    const xPosChevron = xRight - iframeChevron.offsetWidth;
+    const yPosChevron = yPos + offsetY;
 
-    for (let promptParagraphString of htmlBadgeFragmentPromptParagraphStrings) {
-      const paragraph = document.createElement("p");
-      paragraph.appendChild(document.createTextNode(promptParagraphString));
-      htmlBadgeFragmentPromptContents.appendChild(paragraph);
+    iframeChevron.style.marginLeft = `${xPosChevron}px`;
+    iframeChevron.style.marginTop = `${yPosChevron}px`;
+
+    const calculateOffsetDiff = window.innerWidth - fencePosition.x;
+
+    // Flip the iframe to show on the left side when icon is too close to the edge
+    if (iframePaddingAllowance > calculateOffsetDiff) {
+      iframeBox.style.marginLeft = `${xLeft}px`;
+      iframeChevron.classList.add("fbc-chevron-arrow-right");
+      iframeChevron.style.marginLeft = `${xPosChevron - fencePos.offsetWidth - iframeChevron.offsetWidth - offsetX}px`;
     }
 
-    htmlBadgeFragmentPromptDiv.appendChild(htmlBadgeFragmentPromptContents);
+    else {
+      iframeChevron.classList.remove("fbc-chevron-arrow-right");
+    }
 
-    const htmlBadgeFragmentPromptButtonDiv = document.createElement("div");
-    htmlBadgeFragmentPromptButtonDiv.className = "fbc-badge-prompt-buttons";
+    iframeChevron.classList.remove("fbc-chevron-arrow-top");
+  }
 
-    for (let buttonString of htmlBadgeFragmentPromptButtonStrings) {
-      const button = document.createElement("button");
-      button.className = "fbc-badge-prompt-" + buttonString;
-      button.appendChild(document.createTextNode( browser.i18n.getMessage(buttonString) ));
-      htmlBadgeFragmentPromptButtonDiv.appendChild(button);
+  // Mobile Orientation
+  else {
```

Comment:
suggestion (non-blocking): whenever possible I like to refactor `if () { ... } else { ... }` code into `if () { func(); return;} func2();` So here it would be something like:

```js
const OFFSET_X = 20;
const OFFSET_Y = 55;

function desktopOrientation(iframeBox, iframeChevron, fencePos, fencePosition, iframePaddingAllowance) {
  // Desktop Values
  const xLeft = fencePosition.x - iframePaddingAllowance;
  const xRight = fencePosition.x + OFFSET_X + fencePos.offsetWidth;
  const yPos = fencePosition.y - OFFSET_Y;

  // Position iframe relative to FBC Icon
  iframeBox.style.marginLeft = `${xRight}px`;
  iframeBox.style.marginTop = `${yPos}px`;

  // Add Chevron (Default left arrow)
  const xPosChevron = xRight - iframeChevron.offsetWidth;
  const yPosChevron = yPos + OFFSET_Y;

  iframeChevron.style.marginLeft = `${xPosChevron}px`;
  iframeChevron.style.marginTop = `${yPosChevron}px`;

  const calculateOffsetDiff = window.innerWidth - fencePosition.x;

  iframeChevron.classList.remove("fbc-chevron-arrow-top");

  // Flip the iframe to show on the left side when icon is too close to the edge
  if (iframePaddingAllowance > calculateOffsetDiff) {
    iframeBox.style.marginLeft = `${xLeft}px`;
    iframeChevron.classList.add("fbc-chevron-arrow-right");
    iframeChevron.style.marginLeft = `${xPosChevron - fencePos.offsetWidth - iframeChevron.offsetWidth - OFFSET_X}px`;
    return;
  }
  iframeChevron.classList.remove("fbc-chevron-arrow-right");
}

function mobileOrientation(iframeElement, iframeChevron, iframeBox, fencePosition){
  // Mobile Values
  const xPosMobile = fencePosition.x;
  const yPosMobile = fencePosition.y + OFFSET_Y;
  for (const panels of iframeElement) {
    panels.width = window.innerWidth;
    if (window.innerWidth > 480) {
      panels.width = 350;
    }
  }

  iframeChevron.classList.add("fbc-chevron-arrow-top");
  iframeBox.style.marginTop = `${yPosMobile}px`;

  const xPosChevronMobile = xPosMobile;
  const yPosChevronMobile = yPosMobile - iframeChevron.offsetWidth;

  iframeChevron.style.marginLeft = `${xPosChevronMobile}px`;
  iframeChevron.style.marginTop = `${yPosChevronMobile}px`;
}

function positionIframe(fencePos) {
  const fencePosition = fencePos.getBoundingClientRect();
  const iframeBox = document.querySelector(".fbc-content-box");
  const iframeWrapper = document.querySelector(".fbc-wrapper");
  const iframeElement = iframeWrapper.getElementsByTagName("iframe");
  const iframeChevron = document.querySelector(".fbc-iframe-chevron");

  const iframePaddingAllowance = iframeBox.offsetWidth + OFFSET_X;

  const iconRightAllowance = window.innerWidth - fencePosition.x + fencePos.offsetWidth;
  const iconLeftAllowance = window.innerWidth - iconRightAllowance;

  if (iconRightAllowance > iframePaddingAllowance || iconLeftAllowance > iframePaddingAllowance) {
    desktopOrientation(iframeBox, iframeChevron, fencePos, fencePosition, iframePaddingAllowance);
  }
  mobileOrientation(iframeElement, iframeChevron, iframeBox, fencePosition);
}
```

Reply:
Refactored.

Reply:
```js
function positionIframe(fencePos) {
  const fencePosition = fencePos.getBoundingClientRect();
  const iframeBox = document.querySelector(".fbc-content-box");
  const iframeWrapper = document.querySelector(".fbc-wrapper");
  const iframeElement = iframeWrapper.getElementsByTagName("iframe");
  const iframeChevron = document.querySelector(".fbc-iframe-chevron");

  const offsetX = 20;
  const offsetY = 55;

  const iframePaddingAllowance = iframeBox.offsetWidth + offsetX;

  const iconRightAllowance = window.innerWidth - fencePosition.x + fencePos.offsetWidth;
  const iconLeftAllowance = window.innerWidth - iconRightAllowance;

  if (iconRightAllowance > iframePaddingAllowance || iconLeftAllowance > iframePaddingAllowance) {
    return desktopOrientation(iframeBox, iframeChevron, offsetY, fencePosition, iframePaddingAllowance, fencePos, offsetX);
  }
  return mobileOrientation(iframeElement, iframeChevron, iframeBox, fencePosition, offsetY);
}

function mobileOrientation(iframeElement, iframeChevron, iframeBox, fencePosition, offsetY){
  // Mobile Values
  const xPosMobile = fencePosition.x;
  const yPosMobile = fencePosition.y + offsetY;

  for (const panels of iframeElement) {
    panels.width = window.innerWidth;
    if (window.innerWidth > 480) {
      panels.width = 350;
    }
  }

  iframeChevron.classList.add("fbc-chevron-arrow-top");
  iframeBox.style.marginTop = `${yPosMobile}px`;

  const xPosChevronMobile = xPosMobile;
  const yPosChevronMobile = yPosMobile - iframeChevron.offsetWidth;

  iframeChevron.style.marginLeft = `${xPosChevronMobile}px`;
  iframeChevron.style.marginTop = `${yPosChevronMobile}px`;

}

function desktopOrientation(iframeBox, iframeChevron, offsetY, fencePosition, iframePaddingAllowance, fencePos, offsetX) {
  // Desktop Values
  const xRight = fencePosition.x + offsetX + fencePos.offsetWidth;
  const xLeft = fencePosition.x - iframePaddingAllowance;
  const yPos = fencePosition.y - offsetY;

  // Position iframe relative to FBC Icon
  iframeBox.style.marginLeft = `${xRight}px`;
  iframeBox.style.marginTop = `${yPos}px`;

  // Add Chevron (Default left arrow)
  const xPosChevron = xRight - iframeChevron.offsetWidth;
  const yPosChevron = yPos + offsetY;

  iframeChevron.style.marginLeft = `${xPosChevron}px`;
  iframeChevron.style.marginTop = `${yPosChevron}px`;

  const calculateOffsetDiff = window.innerWidth - fencePosition.x;

  // Flip the iframe to show on the left side when icon is too close to the edge
  if (iframePaddingAllowance > calculateOffsetDiff) {
    iframeBox.style.marginLeft = `${xLeft}px`;
    iframeChevron.classList.add("fbc-chevron-arrow-right");
    iframeChevron.style.marginLeft = `${xPosChevron - fencePos.offsetWidth - iframeChevron.offsetWidth - offsetX}px`;
  }
  else {
    iframeChevron.classList.remove("fbc-chevron-arrow-right");
  }
  iframeChevron.classList.remove("fbc-chevron-arrow-top");
}
```

I refactored it again. How is this?

Reply:
Looks good. There's one more `else {}` in `desktopOrientation` there that could be turned into this?

```js
  // Flip the iframe to show on the left side when icon is too close to the edge
  iframeChevron.classList.remove("fbc-chevron-arrow-top");
  if (iframePaddingAllowance > calculateOffsetDiff) {
    iframeBox.style.marginLeft = `${xLeft}px`;
    iframeChevron.classList.add("fbc-chevron-arrow-right");
    iframeChevron.style.marginLeft = `${xPosChevron - fencePos.offsetWidth - iframeChevron.offsetWidth - offsetX}px`;
    return;
  }
  iframeChevron.classList.remove("fbc-chevron-arrow-right");
```

Reply:
done!